src/Nest/QueryDsl/TermLevel/Range/RangeQueryJsonConverter.cs

```diff
@@ -26,25 +26,40 @@ public override object ReadJson(JsonReader reader, Type objectType, object exist
 			var jo = firstProp.Value.Value<JObject>();
 			if (jo == null) return null;
 
+			var fq = GetRangeQuery(serializer, jo);
 
-			var isNumeric = !jo.Properties().Any(p=>p.Name == "format" || p.Name == "time_zone")
-				&& jo.Properties().Any(p=> _rangeKeys.Contains(p.Name) && (p.Value.Type  == JTokenType.Integer || p.Value.Type == JTokenType.Float));
+			fq.Name = GetPropValue<string>(jo, "_name");
+			fq.Boost = GetPropValue<double?>(jo, "boost");
+			fq.Field = field;
+
+			return fq;
+		}
+
+		private static IRangeQuery GetRangeQuery(JsonSerializer serializer, JObject jo)
+		{
+			var isLong = !jo.Properties().Any(p => p.Name == "format" || p.Name == "time_zone")
+			             && jo.Properties().Any(p => _rangeKeys.Contains(p.Name) && p.Value.Type == JTokenType.Integer);
 
+			var isNumeric = !jo.Properties().Any(p => p.Name == "format" || p.Name == "time_zone")
```

Comment:
is it possible to refactor this to remove the duplicated code within `isLong` and `isNumeric` checks. It may be possible to reduce down to one enumeration/iteration of `jo.Properties` to determine `IRangeQuery`